src/tools/schema2kotlin.ts

```diff
@@ -26,10 +26,10 @@ const keywords = [
 ];
 
 const typeMap = {
-  'T': {type: 'String?', decodeFn: 'decodeText()'},
-  'U': {type: 'String?', decodeFn: 'decodeText()'},
-  'N': {type: 'Double?', decodeFn: 'decodeNum()'},
-  'B': {type: 'Boolean?', decodeFn: 'decodeBool()'},
+  'T': {type: 'String', decodeFn: 'decodeText()'},
+  'U': {type: 'String', decodeFn: 'decodeText()'},
+  'N': {type: 'Double', decodeFn: 'decodeNum()'},
+  'B': {type: 'Boolean', decodeFn: 'decodeBool()'},
```

Comment:
Disclaimer: This may be a bad idea, but...

What if we had non-user specified default values? Something like we [used to have](https://github.com/PolymerLabs/arcs/commit/28e274f8ac9f975a448d910a54a3736ba94e5aac#diff-722134b61385e9507e60937e4bbd9015).

There are some pros/cons with this: 
Pro:
- We'd minimize repetition on the behalf of the developer
- developers still get the flexibility to choose their own default values by passing in arguments at construction time

Con: 
- We fail the Wasm API contract (see [here](https://github.com/PolymerLabs/arcs/blob/fd2a992cab32996d25361033b097fc4b91e69558/src/wasm/kotlin/javatests/arcs/EntityClassApiTest.kt#L31) and [here](https://github.com/PolymerLabs/arcs/blob/fd2a992cab32996d25361033b097fc4b91e69558/src/wasm/cpp/tests/entity-class-test.cc#L42))
- Is a bad model of what's happening in the runtime (values should be "not set" in the same way as JavaScript, not initialized to default values). 

Response to Con.2: Isn't this inevitable, anyway, if we're going the route of having developers specify default values? In that case, shouldn't we make it easier for devs?

Thoughts @SHeimlich @mykmartin?

Reply:
The Wasm API tests aren't really a contract; they're under our control and we can just update to match the expected behaviour. I don't think we should have user-specified default values, because that's really just optional fields with extra steps.

So, given a JS entity object with a missing "required" field, we can:
a) create a valid wasm entity and populate the field with the type-appropriate zero value.
b) throw an error, because that field's required, yo.

I think (b) is the correct approach (and will eventually be applied to the JS runtime as well), but I'm ok with (a) as an interim solution to smooth the transition.